src/main/java/pl/touk/parking/owner/ParkingOwnerController.java

```diff
@@ -0,0 +1,41 @@
+package pl.touk.parking.owner;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import pl.touk.parking.ParkingException;
+import pl.touk.parking.domain.Money;
+import pl.touk.parking.web.security.SecurityRoles;
+
+import javax.annotation.security.RolesAllowed;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+@RestController
+public class ParkingOwnerController {
+
+    private final ParkingOwnerService ownerService;
+
+    @Autowired
+    public ParkingOwnerController(ParkingOwnerService ownerService) {
+        this.ownerService = ownerService;
+    }
+
+    @GetMapping("income")
+    @RolesAllowed(SecurityRoles.OWNER)
+    public BigDecimal calculateIncome(@RequestParam String date) {
```

Comment:
czemu to od razu nie jest LocalDate?

Reply:
Można użyć adnotacji `@DateTimeFormat`, ale i tak musiałbym dodać obsługę niepoprawnego formatu daty.